packages/gasket-plugin-logger/src/index.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable no-console */
+/* eslint-disable spaced-comment */
+
+///<reference types="@gasket/plugin-command"/>
```

Comment:
I'd be interested in understanding what downsides there are to authoring in TypeScript. We frequently have to fix up `.d.ts` files because they're disconnected from the actual code. Generating the `.d.ts` from TypeScript is a better way to ensure that they're correct. Also, having TypeScript validation of usage of core plugins is better than a developer honor system.

> adds some unnecessary churn for developing core plugins

Do you have an example of some of this churn?

Reply:
The biggest benefit is having the code that you write being the code that is executed. WIth TS, a build step is necessary to transpile to working JS. When developing locally, sure you can set up a build watcher with plugin code linked into an app. That helps, but it does not if you're trying work with a debugger. It's natural to set breakpoints in your working code, not transpiled code. Additionally, debugging or just reading the executed code is helpful if you're trying to help someone debug a problem over a call. If it's in TypeScript, going to a function definition takes you to a `.d.ts` file which is usually not helpful.

As far as churn, we struggled with the TS configs in two different monorepos with apps and plugins. Maybe friction is a better term for it. It's hard to point you to an example without an active repro case now as we did get it sorted out. It wasn't fun, though, and I'd prefer to avoid it.

I do appreciate types. Don't get me wrong. However, TypeScript types of core plugins are only helpful if the consuming app is using them, with TypeScript otherwise getting in the way during debugging and linking in development. I think there is some middle ground of using JSDocs with TypeScript for our core plugins alluded to in [this article](https://devclass.com/2023/05/11/typescript-is-not-worth-it-for-developing-libraries-says-svelte-author-as-team-switches-to-javascript-and-jsdoc/).